components/ai_chat/core/browser/test_utils.cc

```diff
@@ -208,6 +210,44 @@ void ExpectConversationEntryEquals(base::Location location,
   }
 }
 
+void ExpectToolUseEventEquals(base::Location location,
+                              const mojom::ToolUseEventPtr& a,
+                              const mojom::ToolUseEventPtr& b) {
+  SCOPED_TRACE(testing::Message() << location.ToString());
+  if (!a || !b) {
+    EXPECT_EQ(a, b);  // Both should be null or neither
+    return;
+  }
+  EXPECT_EQ(a->tool_id, b->tool_id);
+  EXPECT_EQ(a->tool_name, b->tool_name);
+  EXPECT_EQ(a->input_json, b->input_json);
+  EXPECT_EQ(a->output.has_value(), b->output.has_value());
+  if (a->output.has_value() || b->output.has_value()) {
+    EXPECT_EQ(a->output->size(), b->output->size());
+    for (size_t i = 0; i < a->output->size(); ++i) {
+      SCOPED_TRACE(testing::Message()
+                   << "Comparing tool output at index " << i);
+      const auto& a_output = a->output->at(i);
+      const auto& b_output = b->output->at(i);
+      EXPECT_EQ(a_output->which(), b_output->which());
```

Comment:
I think this should maybe be an `ASSERT_EQ` as stuff will crash if it isn't equal